src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80122EEC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80122F28.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80122F9C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80122FCC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_8012300C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_8012301C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123140.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123358.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/Player_InCsMode.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123420.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123434.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123448.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801234B0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801234D4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80123590.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801235DC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_8012364C.s")
+#include "objects/object_mask_meoto/object_mask_meoto.h"
+
+typedef struct {
+    /* 0x00 */ Vec3f unk_00;
+    /* 0x0C */ Vec3f unk_0C;
+    /* 0x18 */ s16 unk_18;
+    /* 0x1A */ s16 unk_1A;
+} struct_801F58B0; // size = 0x1C
+
+extern struct_801F58B0 D_801F58B0[3][3];
+
+extern s32 D_801F59C8[2];
+
+extern s32 D_801F59E0;
+
+extern Vec3f D_801F59E8;
+
+void func_80127B64(struct_801F58B0 arg0[], UNK_TYPE arg1, Vec3f* arg2);
+
+s32 func_801226E0(GlobalContext* globalCtx, s32 arg1) {
+    if (arg1 == 0) {
+        func_80169E6C(globalCtx, 0, 0xBFF);
+        if (globalCtx->sceneNum == SCENE_KAKUSIANA) {
+            return 1;
+        }
+    }
+
+    gSaveContext.respawn[0].data = 0;
+    return arg1;
+}
+
+s32 func_80122744(GlobalContext* globalCtx, struct_80122744_arg1* arg1, u32 arg2, Vec3s* arg3) {
+    arg1->unk_00 = arg2;
+    arg1->unk_01 = 0;
+    arg1->unk_04 = arg3;
+    return 1;
+}
+
+s32 func_80122760(GlobalContext* globalCtx, struct_80122744_arg1* arg1, f32 arg2) {
+    if (arg1->unk_01 < arg1->unk_00) {
+        Player* player = GET_PLAYER(globalCtx);
+        Vec3f sp30;
+        s32 pad;
+        s16 yaw;
+        f32 distXZ;
+
+        Math_Vec3s_ToVec3f(&sp30, &arg1->unk_04[arg1->unk_01]);
+        yaw = Math_Vec3f_Yaw(&player->actor.world.pos, &sp30);
+        func_800B6F20(globalCtx, &globalCtx->actorCtx.unk_26C, arg2, yaw);
+        globalCtx->actorCtx.unk268 = 1;
+        distXZ = Math_Vec3f_DistXZ(&player->actor.world.pos, &sp30);
+
+        if ((fabsf(player->actor.world.pos.y - sp30.y) < 50.0f) && (distXZ < arg2)) {
+            arg1->unk_01++;
+        }
+
+        return 0;
+    }
+
+    return 1;
```

Comment:
```suggestion
        return false;
    }

    return true;
```
This function seems to be a boolean return